core/runtime/jsruntime.rs

```diff
@@ -1240,89 +1271,123 @@ impl JsRuntime {
     )
   }
 
+  /// Call a function and returns a future resolving with the return value of the
+  /// function. If the function returns a promise, the future will resolve only once the
+  /// event loop resolves the underlying promise. If the future rejects, the future will
+  /// resolve with the underlying error.
+  ///
+  /// The event loop must be polled seperately for this future to resolve. If the event loop
+  /// is not polled, the future will never make progress.
+  pub fn call(
+    &mut self,
+    function: &v8::Global<v8::Function>,
+  ) -> impl Future<Output = Result<v8::Global<v8::Value>, Error>> {
+    self.call_with_args(function, &[])
+  }
+
+  /// Call a function and returns a future resolving with the return value of the
+  /// function. If the function returns a promise, the future will resolve only once the
+  /// event loop resolves the underlying promise. If the future rejects, the future will
+  /// resolve with the underlying error.
+  ///
+  /// The event loop must be polled seperately for this future to resolve. If the event loop
+  /// is not polled, the future will never make progress.
+  pub fn scoped_call(
+    scope: &mut v8::HandleScope,
```

Comment:
Not a fan of this name - maybe `call_with_scope`/`call_in_scope`? What's the use case for it? Just to pass scope explicitly?

Reply:
I couldn't really come up with a better name, `call_with_scope` might work -- it allows you to make a call without needing a `&mut self` for the `JsRuntime`. To be honest, I expect to pull most of these methods off into some sort of `JsRuntimeHandle` but that needs a bit of experimentation work.